DeratControl.Domain/Entities/Facility.cs

```diff
@@ -17,5 +19,21 @@ private Facility()
             this.Perimeters = new List<Perimeter>();
             this.Reviews = new List<Review>();
         }
+
+        public void RemovePerimeter(PerimeterType perimeterType)
```

Comment:
sorry for the dumb question, but did you try to compile this code?
RemovePerimeter returns void but you use 
return perimeterToRemove; - statement, is it legit?